### PR TITLE
BUG: Spurious matches in DataFrame.duplicated when keep=False

### PR DESCRIPTION
**Pull request by evanpw**
Fixes GH #11864

### CODE REVIEW

File: pandas/tests/test_frame.py [hunk 1]
```diff
@@ -8510,6 +8510,11 @@ def test_drop_duplicates(self):
         df = pd.DataFrame([[-x, x], [x, x + 4]])
         assert_frame_equal(df.drop_duplicates(), df)
 
+        # GH 11864
+        df = pd.DataFrame([i] * 9 for i in range(16))
+        df = df.append([[1] + [0] * 8], ignore_index=True)
```
Comment (jreback) [inline pandas/tests/test_frame.py]: can u run thru all of the keep options in a loop here

**Comment by jreback** [comment]
lgtm - ping when updated and green

**Comment by jreback** [comment]
@evanpw can you rebase / update

**Comment by evanpw** [comment]
Added test loop, rebased, travis is green.

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/b431f85a505a4d84e38f4d6f9ed5080775d4f31c


thanks!

**Comment by zhbain** [comment]
Python 3.6.4 :: Anaconda, Inc.
pandas Version: 0.22.0
i still get the bug when the data has 10million rows
but 10 rows was ok.
the dtype is both int64.

**Pull request closed without merging**